File: No_1975/Maximum Matrix Sum.cpp
class Solution {
public:
    long long maxMatrixSum(vector<vector<int>>& matrix) {
        int negativeCount = 0;
        int minAbs = INT_MAX;
        long long positiveSum = 0;
        int m = matrix.size(), n = matrix[0].size();
        for(int i=0; i<m; i++) {
            for(int j=0; j<n; j++) {
                int element = matrix[i][j];
                if(element < 0) {
                    negativeCount++;
                }
                if(abs(element) < minAbs) {
                    minAbs = abs(element);
                }
                positiveSum += abs(element);
            }
        }
        
        if(negativeCount % 2 == 0) return positiveSum;
        return positiveSum - 2*minAbs;
    }
};

File: No_1975/Maximum Matrix Sum.py
class Solution:
    def maxMatrixSum(self, matrix: List[List[int]]) -> int:
        total_abs_sum = 0
        min_abs = float("inf")
        negative_count = 0

        for row in matrix:
            for val in row:
                abs_val = abs(val)
                total_abs_sum += abs_val
                min_abs = min(min_abs, abs_val)
                if val < 0:
                    negative_count += 1

        if negative_count % 2 == 0:
            return total_abs_sum
        return total_abs_sum - 2 * min_abs
